<launch>

<!-- The following is the table for a (2, 0) robot inspired from the MOBRO course
______________________________________________________________________________________
| Wheel |   L   |  alpha  |   d   |   beta    |  gamma  |   phi   |      psi         |
______________________________________________________________________________________
|  1f   |   a   | -M_PI/2 |   0   |    0      |  M_PI/2 | phi_1f  |       0          |
|  2f   |   a   |  M_PI/2 |   0   |    0      | -M_PI/2 | phi_2f  |       0          |
|  3c   |   b   |  M_PI   |   c   |  beta_3c  |    0    | phi_3c  | M_PI + beta_3c   |

You are allowed to change the length of the robot part. You should not try to change
the angles and orientation of the robot as all the kinematic will change. Moreover,
it does not make much sense to change the robot parts orientation with respect
to the moving platform frame                                                        -->

  <arg name="a" value="0.2"/>
  <arg name="b" value="0.4"/>
  <arg name="c" value="0.08"/>

  <arg name="r" value="0.05"/>
  <arg name="w_max" value="10.0"/>

  <arg name="x_spacing" value="0.2"/>
  <arg name="y_spacing" value="0.2"/>

  <arg name="file_name" value="completo"/>
  <arg name="folder_name" value="recorded_data"/>

  <!-- joy node -->
  <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
    <param name="dev" type="string" value="/dev/input/js1" />
  </node>

  <!-- The compute_twist adapter -->
  <node respawn="true" pkg="simulation" type="simulation_compute_twist" name="compute_twist" output="screen">
  </node>

  <group ns="simulation">

    <node pkg="simulation" type="simulation_file_handler" name="file_handler" args="$(find data)/$(arg folder_name)  $(arg file_name)" output="screen">
    </node>

    <!-- The robot_2_0 -->
    <node pkg="simulation" type="simulation_robot_2_0" name="robot_2_0" output="screen">
      <param name="track_gauge"         value="$(arg a)"      />
      <param name="wheel_radius"        value="$(arg r)"      />
      <param name="joint_offset"        value="$(arg b)"      />
      <param name="castor_arm"          value="$(arg c)"      />
      <param name="actuator_max_speed"  value="$(arg w_max)"  />
    </node>

    <!-- Launch Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulation)/rviz/first_save.rviz">
    </node>
  </group>

  <param name="robot_description" textfile="$(find simulation_description)/urdf/robot_2_0.urdf" />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
  </node>


</launch>
